src/code/z_demo.c

```diff
@@ -1232,684 +1404,776 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
+            case CS_TRANS_GREEN_UNFILL:
                 play->envCtx.screenFillColor[0] = 0;
                 play->envCtx.screenFillColor[1] = 255;
                 play->envCtx.screenFillColor[2] = 0;
-                if (cmd->base == 4) {
-                    play->envCtx.screenFillColor[3] = (1.0f - temp) * 255.0f;
+
+                if (cmd->type == CS_TRANS_GREEN_FILL) {
+                    play->envCtx.screenFillColor[3] = (1.0f - lerp) * 255.0f;
                 } else {
-                    play->envCtx.screenFillColor[3] = 255.0f * temp;
+                    play->envCtx.screenFillColor[3] = 255.0f * lerp;
                 }
                 break;
 
-            case 9:
+            case CS_TRANS_TRIGGER_INSTANCE:
                 gSaveContext.cutsceneTransitionControl = 1;
                 break;
 
-            case 10:
-            case 11:
+            case CS_TRANS_BLACK_FILL:
+            case CS_TRANS_BLACK_UNFILL:
                 play->envCtx.screenFillColor[0] = 0;
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 0;
-                if (cmd->base == 10) {
-                    play->envCtx.screenFillColor[3] = (1.0f - temp) * 255.0f;
+
+                if (cmd->type == CS_TRANS_BLACK_FILL) {
+                    play->envCtx.screenFillColor[3] = (1.0f - lerp) * 255.0f;
                 } else {
-                    play->envCtx.screenFillColor[3] = 255.0f * temp;
+                    play->envCtx.screenFillColor[3] = 255.0f * lerp;
                 }
                 break;
 
-            case 12:
-                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * temp);
+            case CS_TRANS_BLACK_HALF_FILL:
+                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * lerp);
                 break;
 
-            case 13:
+            case CS_TRANS_BLACK_HALF_UNFILL:
                 play->envCtx.screenFillColor[0] = 0;
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 0;
-                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - temp) * 155.0f);
+                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - lerp) * 155.0f);
                 break;
         }
     }
 }
 
-// Command 0x1 & 0x5: Camera Eye Points
-s32 Cutscene_Command_CameraEyePoints(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 relativeToLink) {
-    s32 shouldContinue = 1;
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_UpdateCamEyeSpline(PlayState* play, CutsceneContext* csCtx, u8* script, u8 relativeToPlayer) {
+    s32 shouldContinue = true;
+    CsCmdGeneric* cmd = (CsCmdGeneric*)script;
     s32 size;
 
-    cmd += 8;
-    size = 8;
+    script += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
+
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((csCtx->camEyeSplinePointsAppliedFrame < cmd->startFrame) ||
```

Comment:
This is somewhat subjective but I find
```suggestion
        ((cmd->startFrame > csCtx->camEyeSplinePointsAppliedFrame) ||
```
easier to parse than the reverse

this matches for the other spots like this (four in total including this one)